src/main/java/com/google/devtools/build/lib/skyframe/PlatformLookupUtil.java

```diff
@@ -66,6 +85,53 @@
     return platforms;
   }
 
+  /** Validate that all keys are for actual platform targets. */
+  private static void validatePlatformKeys(
+      ImmutableList<ConfiguredTargetKey> platformKeys, Environment env)
+      throws InterruptedException, InvalidPlatformException {
+    // Load the packages. This should already be in Skyframe and thus not require a restart.
+    Map<ConfiguredTargetKey, PackageIdentifier> targetsToPackageIdentifiers =
+        platformKeys.stream()
+            .distinct()
+            .collect(
+                toImmutableMap(Function.identity(), ctk -> ctk.getLabel().getPackageIdentifier()));
+    ImmutableSet<PackageValue.Key> packageKeys =
+        targetsToPackageIdentifiers.values().stream()
+            .distinct()
+            .map(PackageValue::key)
+            .collect(toImmutableSet());
+
+    Map<PackageIdentifier, Package> packages = new HashMap<>();
+    Map<SkyKey, ValueOrException<NoSuchPackageException>> values =
+        env.getValuesOrThrow(packageKeys, NoSuchPackageException.class);
+    if (env.valuesMissing()) {
+      return;
+    }
+    for (Map.Entry<SkyKey, ValueOrException<NoSuchPackageException>> value : values.entrySet()) {
+      try {
+        PackageValue packageValue = (PackageValue) value.getValue().get();
+        packages.put(packageValue.getPackage().getPackageIdentifier(), packageValue.getPackage());
+      } catch (NoSuchPackageException e) {
```

Comment:
It might be useful to add a comment here why this exception cannot happen/can safely be ignored.

Reply:
Added error handling.